WordPress/Classes/ViewRelated/Reader/ReaderBrowseSiteViewController.m

```diff
@@ -78,10 +100,50 @@ - (void)willAnimateRotationToInterfaceOrientation:(UIInterfaceOrientation)toInte
 
 #pragma mark - Configuration
 
-- (void)createSiteTopic
+- (void)getTopicFromSiteID
 {
+    [self showLoadingSite];
+    __weak __typeof(self) weakSelf = self;
     ReaderTopicService *topicService = [[ReaderTopicService alloc] initWithManagedObjectContext:[[ContextManager sharedInstance] mainContext]];
-    self.siteTopic = [topicService siteTopicForPost:self.post];
+    [topicService siteTopicForSiteWithID:self.siteID
+                                 success:^(NSManagedObjectID *objectID, BOOL isFollowing) {
+                                     weakSelf.isFollowing = isFollowing;
+                                     NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+                                     NSError *error;
+                                     weakSelf.siteTopic = (ReaderTopic *)[context existingObjectWithID:objectID error:&error];
```

Comment:
We may need some error handling here. If the `ReaderTopic` is nil at this point, do we still wanna proceed?